tools/ci/src/main.rs

```diff
@@ -1,44 +1,99 @@
 use xshell::{cmd, pushd};
 
+use bitflags::bitflags;
+
+bitflags! {
+    struct Check: u32 {
+        const FORMAT = 0b00000001;
+        const CLIPPY = 0b00000010;
+        const COMPILE_FAIL = 0b00000100;
+        const TEST = 0b00001000;
+        const DOC_TEST = 0b00010000;
+        const DOC_CHECK = 0b00100000;
+        const BENCH_CHECK = 0b01000000;
+        const EXAMPLE_CHECK = 0b10000000;
+    }
+}
+
 fn main() {
     // When run locally, results may differ from actual CI runs triggered by
     // .github/workflows/ci.yml
     // - Official CI runs latest stable
     // - Local runs use whatever the default Rust is locally
 
-    // See if any code needs to be formatted
-    cmd!("cargo fmt --all -- --check")
-        .run()
-        .expect("Please run 'cargo fmt --all' to format your code.");
+    let what_to_run = match std::env::args().nth(1).as_deref() {
+        Some("format") => Check::FORMAT,
+        Some("clippy") => Check::CLIPPY,
+        Some("compile-fail") => Check::COMPILE_FAIL,
+        Some("test") => Check::TEST,
+        Some("doc-test") => Check::DOC_TEST,
+        Some("doc-check") => Check::DOC_CHECK,
+        Some("bench-check") => Check::BENCH_CHECK,
+        Some("example-check") => Check::EXAMPLE_CHECK,
+        Some("lints") => Check::FORMAT | Check::CLIPPY,
+        Some("doc") => Check::DOC_TEST | Check::DOC_CHECK,
+        Some("compile") => Check::COMPILE_FAIL | Check::BENCH_CHECK | Check::EXAMPLE_CHECK,
+        _ => Check::all(),
+    };
 
-    // See if clippy has any complaints.
-    // - Type complexity must be ignored because we use huge templates for queries
-    cmd!("cargo clippy --workspace --all-targets --all-features -- -D warnings -A clippy::type_complexity -W clippy::doc_markdown")
+    if what_to_run.contains(Check::FORMAT) {
+        // See if any code needs to be formatted
+        cmd!("cargo fmt --all -- --check")
+            .run()
+            .expect("Please run 'cargo fmt --all' to format your code.");
```

Comment:
I think the `--check` should only be used by the CI, and on local use, it should directly format the code.

Reply:
The goal of this script is to run CI checks and help you run them locally, not to fix it for you. Maybe we could add another flag for that later, but that's out of scope of this PR